src/Microsoft.Graph.Core/Tasks/PageIterator.cs

```diff
@@ -0,0 +1,269 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /*
+     Spec https://github.com/microsoftgraph/msgraph-sdk-design/blob/master/tasks/PageIteratorTask.md
+    */
+
+    /// <summary>
+    /// Use PageIterator&lt;TEntity&gt; to automatically page through result sets across multiple calls 
+    /// and process each item in the result set.
+    /// </summary>
+    /// <typeparam name="TEntity">The Microsoft Graph entity type returned in the result set.</typeparam>
+    public class PageIterator<TEntity>
+    {
+        private IBaseClient _client;
```

Comment:
readonly?

Reply:
_client is also settable via the static method `CreatePageIterator`. For it to be readonly it would only be set by the constructor.